src/components/MultiStepIndicator/components/StepIndicator/StepIndicator.jsx

```diff
@@ -90,9 +95,11 @@ const StepIndicator = ({
   return (
     <li
       className={cx(...getClassNamesWithSuffix(""), stepComponentClassName, {
-        [baseClassNameWithAnimation]: statusChangeAnimationState
+        [baseClassNameWithAnimation]: statusChangeAnimationState,
+        clickable: onClick
       })}
       aria-label={ariaLabel}
+      onClick={handleClick}
```

Comment:
Once an element is clickable, there are a few changes it needs in order to be keyboard-accessible:

- An `onKeyPress` handler (you can use the existing `useKeyEvent` hook to get it easily) to catch 'Enter' and 'Space' key presses, and apply the same `handleClick` function.
- A `tabIndex={0}` attribute

After adding it (if you choose to), make sure it looks OK when tabbing, and that keyboard navigation works.